### PR TITLE
*Added driver component

### PR DESCRIPTION
**Pull request by maxonchil**
*Added opportunity to create a truck
*Created Truck schema, validaton, track HARDCODED data
*Create an utilit to get a truck props
*Added 'asign' button and make it functional
Error handle and logs included

### CODE REVIEW

File: homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const log4js = require("log4js");
+const logger = log4js.getLogger();
+const errorHandler = require("../error.handler");
+const { Truck } = require("../../Schemas/truck.schema");
+
+const trucksPatchHandler = async (req, res) => {
+  const { userID, truckID } = req.body;
+
+  try {
+    const clearAssign = await Truck.updateMany(
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js]: Why do you need this variable?
if you just need to call update many without saving the value it returns, you can just call
`await Truck.updateMany(...`

File: homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      { created_by: userID },
+      { $set: { assigned_to: null } }
+    );
+  } catch (error) {
+    return errorHandler(error.message, res);
+  }
+  logger.info("All trucks is not assigned");
+  try {
+    const assignedTruck = await Truck.findByIdAndUpdate(truckID, {
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js]: The same, you do not to write it to variable, `await` and function call will be enough

File: homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+const log4js = require("log4js");
+const logger = log4js.getLogger();
+const errorHandler = require("../error.handler");
+const { Truck } = require("../../Schemas/truck.schema");
+
+const trucksPatchHandler = async (req, res) => {
+  const { userID, truckID } = req.body;
+
+  try {
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js]: Please change this logic as we have discussed:
Find drivers assigned truck, set `assigned_to` to `null`. And then assign the new truck to the driver

File: homework3/semi-uber/Server/api/handlers/userHandlers/userGetHandler.js [hunk 4]
```diff
@@ -6,14 +6,15 @@ const jwt = require("jsonwebtoken");
 const { secret } = config.get("JWT");
 const errorHandler = require("../error.handler");
 const { Load } = require("../../Schemas/load.schema");
+const { Truck } = require("../../Schemas/truck.schema");
 
 async function getCustomData(stats, userID) {
   if (stats === "Shipper") {
     const loads = await Load.find({ created_by: userID });
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/handlers/userHandlers/userGetHandler.js]: Add error handling for this await

File: homework3/semi-uber/Server/api/handlers/userHandlers/userGetHandler.js [hunk 5]
```diff
@@ -6,14 +6,15 @@ const jwt = require("jsonwebtoken");
...
     return { loads };
   } else {
-    // const trucks = await Load.find({ created_by: userID });
-    // return { trucks };
+    const trucks = await Truck.find({ created_by: userID });
```
Comment (MaxHouz) [inline homework3/semi-uber/Server/api/handlers/userHandlers/userGetHandler.js]: add error handling

Comment (maxonchil) [reply homework3/semi-uber/Server/api/handlers/userHandlers/userGetHandler.js]: Done

File: homework3/semi-uber/Server/api/handlers/userHandlers/userGetHandler.js (hunk 4 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/handlers/userHandlers/userGetHandler.js]: Done

File: homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js (hunk 3 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js]: Done

File: homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js (hunk 2 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js]: Done

File: homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js (hunk 1 above)
Comment (maxonchil) [reply homework3/semi-uber/Server/api/handlers/trucksHandlers/trucksPatchHandler.js]: Done

**Pull request merged**